### PR TITLE
docs: add tedge-watchdog decision page

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Add a new decision page after a decision about the future of the **tedge-watchdog** component and rationale on why shifting focus to independent solutions such as `monit` would lead to a more reliable/holistic approach to moniting components and performing corrective actions to bring a system back to a healthy state.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Bravo555 — APPROVED** [review]

**Review by didier-wenzek — COMMENTED** [review]
I'm okay to go in the direction proposed here. i.e. to stop further development of the **tedge-watchdog** component (at to deprecate it) and to improve the `tedge-monit-setup` for thin-edge.

**Comment by reubenmiller** [comment]
> I'm okay to go in the direction proposed here. i.e. to stop further development of the **tedge-watchdog** component (at to deprecate it) and to improve the `tedge-monit-setup` for thin-edge.

I'm still in two minds about whether to fully deprecate the plugin or not. I'm leaning towards at least keeping it in the code based, so the other users can continue development (if they do prefer the systemd based approach).

@didier-wenzek What do you think?

**Comment by didier-wenzek** [comment]
> I'm still in two minds about whether to fully deprecate the plugin or not.

Do we have known users? Not sure. We have few or no questions about it.

I would keep the code for the `tedge_watchdog` but remove it from the documentation as we do for the `c8y_firmware_plugin`.

**Comment by reubenmiller** [comment]
> > I'm still in two minds about whether to fully deprecate the plugin or not.
> 
> Do we have known users? Not sure. We have few or no questions about it.
> 

We don't have any exact data to support this, however since the tedge-watchdog always needed some manual configuration to enable it, I really doubt anyone has been using it, however I would use caution and expect there to be at least a few users.


> I would keep the code for the `tedge_watchdog` but remove it from the documentation as we do for the `c8y_firmware_plugin`.

The tedge watchdog docs has already been marked as "unlisted" since a while now (when the original limitation was found see https://github.com/thin-edge/thin-edge.io/pull/2635). Unlisted pages don't turn up in the doc searches, but can still be accessed if someone knows the url (e.g. they have saved it in a bookmark) - [here is a link to the unlisted page](https://thin-edge.github.io/thin-edge.io/operate/monitoring/systemd-watchdog/)

**Comment by didier-wenzek** [comment]
> The tedge watchdog docs has already been marked as "unlisted"

So, let's leave things as they are now.

File: design/decisions/0005-tedge-watchdog.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# tedge-watchdog
+
+* Date: __2024-10-18__
+* Status: __New__
```
Comment (didier-wenzek) [inline design/decisions/0005-tedge-watchdog.md:4]:
```suggestion
* Status: __Approved__
```

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by albinsuresh — APPROVED** [review]

File: design/decisions/0005-tedge-watchdog.md (hunk 1 above)
Comment (reubenmiller) [reply design/decisions/0005-tedge-watchdog.md]: Resolved by 79a9ab175aef11dea1cb1446707ff2cb56c768c6

**Pull request merged**